### PR TITLE
Refactor basic auth

### PR DESCRIPTION
**Pull request by maximbaz**
- Use a shared tabs.onUpdated instead of subscribing multiple times
- Extract onAuthRequired into a separate global function

@erayd I'll comment inline on a couple of particularly note-worthy points.

### CODE REVIEW

File: chrome/background.js [hunk 1]
```diff
@@ -99,96 +100,86 @@ function onMessage(request, sender, sendResponse) {
   // spawn a new tab with pre-provided credentials
   if (request.action == "launch") {
     chrome.tabs.create({ url: request.url }, function(tab) {
-      var tabLoaded = false;
```
Comment (maximbaz) [inline chrome/background.js]: I didn't see a particular use for `tabLoaded`, given that we discard onAuthListener immediately after tab is loaded. Did you have a point for this that I'm missing?

File: chrome/background.js [hunk 2]
```diff
@@ -99,96 +100,86 @@ function onMessage(request, sender, sendResponse) {
...
-        info
-      ) {
-        if (tabId == tab.id && info.status == "complete") {
-          // remove listeners
-          chrome.tabs.onUpdated.removeListener(tabCompleteListener);
-          chrome.webRequest.onAuthRequired.removeListener(authListener);
-
-          // mark tab as loaded & wipe credentials
-          tabLoaded = true;
-          request.username = request.password = null;
```
Comment (maximbaz) [inline chrome/background.js]: The reference to this object is in this function and now in `onAuthRequired`, as soon as both functions are gone the credentials are not accessible anyway - is there a point in additionally wiping them here?

File: chrome/background.js (hunk 1 above)
Comment (erayd) [reply chrome/background.js]: That can go. I went through a couple of different versions of the bugfix; that was left over from one of them but is no longer applicable.

File: chrome/background.js (hunk 2 above)
Comment (erayd) [reply chrome/background.js]: As above - this was leftover from an earlier iteration of the bugfix; it can go.

**Comment by maximbaz** [comment]
Thanks @erayd for the confirmations, any other feedback before this is merged?

**Comment by erayd** [comment]
@maximbaz No other feedback at this point - I like what you've done, go ahead and merge it :-).

**Pull request merged**